### PR TITLE
Try to split and paralelize tests

### PR DESCRIPTION
**Pull request by juanitorduz**
Closes https://github.com/py-econometrics/pyfixest/issues/605

### CODE REVIEW

File: .github/workflows/ci-tests.yaml [hunk 1]
```diff
@@ -10,13 +10,13 @@ on:
...
   test:
-    name: "Tests"
+    name: "Tests Core"
     runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: false
       matrix:
         os: ["ubuntu-latest"]
-        python-version: ["3.9", "3.10", "3.11", "3.12"]
+        python-version: ["3.9", "3.12"]
```
Comment (juanitorduz) [inline .github/workflows/ci-tests.yaml:19]: For the core tests its enough to test the min and max versions

File: .github/workflows/ci-tests.yaml [hunk 2]
```diff
@@ -49,6 +49,46 @@ jobs:
...
         uses: codecov/codecov-action@v4
 
+  test_slow:
+    name: "Test Slow"
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ["ubuntu-latest"]
+        python-version: ["3.12"]
```
Comment (juanitorduz) [inline .github/workflows/ci-tests.yaml:59]: I suggest we just use the latest version for the slow tests. WDYT?

Comment (s3alfisc) [reply .github/workflows/ci-tests.yaml:59]: Yes, sounds good to me!

**Review by s3alfisc — APPROVED** [review]

**Comment by juanitorduz** [comment]
ok! Apologies for the random commits above, I was not getting why some slow tests where being shipped but I just realized there are kip statements inside the tests like 

```python
        pytest.skip(
            "Cluster adjustment only works with cluster inference. Nothing to test here."
        )
```
😅

File: pyproject.toml [hunk 3]
```diff
@@ -43,19 +46,20 @@ watermark = "^2.4.3"
...
+  "--strict-markers",
+  "--strict-config",
+  "--cov=pyfixest",
+  "--cov-report=term-missing",
+  "--color=yes",
 ]
 markers = [
-    "extended: mark test to be part of the extended test suite"
+  "slow: mark test to be part of the slow test suite",
+  "extended: mark test to be part of the extended test suite",
```
Comment (juanitorduz) [inline pyproject.toml:58]: In the next iteration, we can split further by adding more markets if necessary.

**Comment by s3alfisc** [comment]
Yep, this here 

```
    if "f3" not in fml:
pytest.skip("No f3 in formula. Nothing to test here.")
```
was the statement that deactivated all IV tests ...

**Pull request merged**